client/src/views/statistics/statistics.ts

```diff
@@ -1,9 +1,102 @@
 import type { Observer } from '@/core/ui/observer';
+import makePercent from '@/core/utils/makePercent';
+import styles from './statistics.module.scss';
 
 export default class Statistics implements Observer {
+  refactorData(state: StoreState): [number, IDonutRecord[]] {
+    const { categories, records } = state;
+    let total = 0;
+    let donutRecord: IDonutRecord[] = [];
+
+    // 지출이 될 수 있는 카테고리로 데이터 초기화
+    categories.forEach((item) => {
+      if (item.type === 'expenditure') {
+        donutRecord.push({ id: item.id, name: item.name, color: item.color, value: 0, percent: 0 });
+      }
+    });
+
+    // 카테고리별 지출값 묶어주기
+    records.forEach((record) => {
+      const isExpenditure = categories[record.categoryId - 1].type === 'expenditure';
+      const category = categories[record.categoryId - 1].name;
+      if (isExpenditure) {
+        total += record.value;
+        const index = donutRecord.findIndex((item) => item.name === category);
+        donutRecord[index].value += record.value;
+      }
+    });
+
+    // 도넛 차트 지출 많은 순으로 정렬
+    donutRecord = donutRecord
+      .filter((item) => item.value !== 0)
+      .sort((a, b) => {
+        if (a.value > b.value) return -1;
+        return 0;
+      });
+
+    // 지출 금액만 따로 뺄 배열 생성
+    const numberArr: number[] = [];
+
+    // numberArr에 퍼센트 추가
+    donutRecord.forEach((item) => {
+      numberArr.push((item.value / total) * 100);
+    });
+
+    // 새로운 배열에 100%에 맞춘 정수 퍼센트 가져오기
+    const newNumberArr = makePercent(numberArr);
+
+    donutRecord.forEach((item, i) => {
```

Comment:
```suggestion
    donutRecord.forEach((_, i) => {
```

사용하지 않는 변수는 `_`로 기입하면 됩니다.

Reply:
아 사용하지 않는 변수 지운다고 지웠는데 또 있었네요.. ㅠㅠ